src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -390,79 +395,27 @@ private void calculateContingencySensitivityValues(PropagatedContingency conting
                         newParticipatingElements = getParticipatingElements(lfNetwork.getBuses(), lfParameters.getBalanceType(), lfParametersExt);
                     }
                 }
-                if (participatingElementsChanged || rhsChanged) {
-                    newFactorStates = calculateFactorStates(loadFlowContext, factorGroups, newParticipatingElements);
-                }
                 // write contingency status
                 resultWriter.writeContingencyStatus(contingency.getIndex(), SensitivityAnalysisResult.Status.SUCCESS);
             } else {
                 // write contingency status
                 resultWriter.writeContingencyStatus(contingency.getIndex(), SensitivityAnalysisResult.Status.NO_IMPACT);
             }
 
+            // we need to recompute the factor states because the rhs has changed or
```

Comment:
Hello @p-arvy , please do not forget to complete this comment

Reply:
Indeed. Ty for the comment @Hadrien-Godard.